src/main/java/org/folio/service/accesstypes/AccessTypesServiceImpl.java

```diff
@@ -59,12 +62,15 @@
   @Override
   public CompletableFuture<AccessTypeCollection> findAll(Map<String, String> okapiHeaders) {
     return repository.findAll(tenantId(okapiHeaders))
+      .thenCombine(mappingService.findAll(okapiHeaders), this::setEachRecordUsage)
```

Comment:
This is not the best way of getting number of usages. When you have thousand of mappings the performance is going down significantly.
You need to perform the calculation on DB side by grouping records and counting the totals

Reply:
Fixed